dingo/utils.py

```diff
@@ -201,3 +201,141 @@ def get_matrices_of_full_dim_polytope(A, b, Aeq, beq):
+        for i in range(0, corr_indices.shape[0]):
+            
+            index1 = corr_indices[i][0]
+            index2 = corr_indices[i][1]
+            
+            reaction1 = reactions[index1]
+            reaction2 = reactions[index2]
+            
+            flux1 = steady_states[index1]
+            flux2 = steady_states[index2]
+                
+            copula = compute_copula(flux1, flux2, cells)
+            rows, cols = copula.shape
+        
+            red_mass = 0
+            blue_mass = 0
+            indicator = 0
+                            
+            for row in range(rows):
+                for col in range(cols):
+                    # values in the diagonal
+                    if ((row-col >= -cop_coeff_1*rows) & (row-col <= cop_coeff_1*rows)):        
+                        # values near the top left and bottom right corner
+                        if ((row+col < cop_coeff_2*rows) | (row+col > cop_coeff_3*rows)):
+                            red_mass = red_mass + copula[row][col]
+                    else:
+                        # values near the top right and bottom left corner
+                        if ((row+col >= cop_coeff_2*rows-1) & (row+col <= cop_coeff_3*rows-1)):
+                            blue_mass = blue_mass + copula[row][col]
+
+            indicator = (red_mass+1e-9) / (blue_mass+1e-9)
+                    
+            # classify specific pair of reactions as positive or negative
+            # correlated based on indicator cutoff 
+            if indicator > indicator_cutoff:
+                pearson = filtered_corr_matrix[index1, index2]
+                indicator_dict[reaction1 + "~" + reaction2] = {'pearson': pearson,
+                                                               'indicator': indicator, 
+                                                               'classification': "positive"}
+
+            elif indicator < 1/indicator_cutoff:
+                pearson = filtered_corr_matrix[index1, index2]
+                indicator_dict[reaction1 + "~" + reaction2] = {'pearson': pearson,
+                                                               'indicator': indicator, 
+                                                               'classification': "negative"}
+
+            # if they do not overcome the cutoff replace their corresponding
+            # value in the correlation matrix with 0
+            else:
+                filtered_corr_matrix[index1, index2] = 0
+                filtered_corr_matrix[index2, index1] = 0
+                pearson = filtered_corr_matrix[index1, index2]
+                indicator_dict[reaction1 + "~" + reaction2] = {'pearson': pearson,
+                                                               'indicator': indicator, 
+                                                               'classification': "no correlation"}
+                
+            print("Completed process of",i+1,"from",corr_indices.shape[0],"copulas")
```

Comment:
All the prints should ideally requested by the user and be off by default.
I'd use a flag `verbose: bool` as an input variable with `False` as default value. Then, print any message only if verbose is true.